presto-main/src/test/java/com/facebook/presto/execution/buffer/TestArbitraryOutputBuffer.java

```diff
@@ -0,0 +1,957 @@
+
+    public static BufferResult getBufferResult(OutputBuffer buffer, OutputBufferId bufferId, long sequenceId, DataSize maxSize, Duration maxWait)
+    {
+        CompletableFuture<BufferResult> future = buffer.get(bufferId, sequenceId, maxSize);
+        return getFuture(future, maxWait);
+    }
+
+    public static BufferResult getFuture(CompletableFuture<BufferResult> future, Duration maxWait)
+    {
+        return tryGetFutureValue(future, (int) maxWait.toMillis(), MILLISECONDS).get();
+    }
+
+    private static ListenableFuture<?> enqueuePage(OutputBuffer buffer, Page page)
+    {
+        ListenableFuture<?> future = buffer.enqueue(page);
+        assertFalse(future.isDone());
+        return future;
+    }
+
+    private static void addPage(OutputBuffer buffer, Page page)
+    {
+        assertTrue(buffer.enqueue(page).isDone(), "Expected add page to not block");
+    }
+
+    private static void assertQueueState(
+            OutputBuffer buffer,
+            int unassignedPages,
+            OutputBufferId bufferId,
+            int bufferedPages,
+            int pagesSent)
+    {
+        OutputBufferInfo outputBufferInfo = buffer.getInfo();
+
+        long assignedPages = outputBufferInfo.getBuffers().stream().mapToInt(BufferInfo::getBufferedPages).sum();
+
+        assertEquals(
+                outputBufferInfo.getTotalBufferedPages() - assignedPages,
+                unassignedPages,
+                "unassignedPages");
+
+        BufferInfo bufferInfo = outputBufferInfo.getBuffers().stream()
+                .filter(info -> info.getBufferId().equals(bufferId))
+                .findAny()
+                .orElse(null);
+
+        assertEquals(
+                bufferInfo,
+                new BufferInfo(
+                        bufferId,
+                        false,
+                        bufferedPages,
+                        pagesSent,
+                        new PageBufferInfo(
+                                bufferId.getId(),
+                                bufferedPages,
+                                sizeOfPages(bufferedPages).toBytes(),
+                                bufferedPages + pagesSent, // every page has one row
+                                bufferedPages + pagesSent)));
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    private static void assertQueueClosed(OutputBuffer buffer, int unassignedPages, OutputBufferId bufferId, int pagesSent)
+    {
+        OutputBufferInfo outputBufferInfo = buffer.getInfo();
+
+        long assignedPages = outputBufferInfo.getBuffers().stream().mapToInt(BufferInfo::getBufferedPages).sum();
+        assertEquals(
+                outputBufferInfo.getTotalBufferedPages() - assignedPages,
+                unassignedPages,
+                "unassignedPages");
+
+        BufferInfo bufferInfo = outputBufferInfo.getBuffers().stream()
+                .filter(info -> info.getBufferId().equals(bufferId))
+                .findAny()
+                .orElse(null);
```

Comment:
Isn't this the same as the helper method: getBufferInfo? Either remove that method or use it here.